### ISSUE TITLE
Improve graph support for custom telemetry vendors

### ISSUE THREAD

**Issue by jshaughn** [kiali/kiali#1679]
The request originates from [this mailing list discussion](https://groups.google.com/d/msgid/kiali-dev/c76df731-9888-46eb-bad7-e7edc8997b25%40googlegroups.com.).

Istio is the reference implementation for graph generation (and all of Kiali) but it may be useful to incrementally support alternate telemetry sources.  This issue is to track and research or work done to further this goal.

**Comment by abonas** [comment]
I think there's a need to refine terminology a bit. 
Telemetry vendor can be interpreted differently - it could be:
1. Prometheus itself, e.g. getting Istio metrics into Kiali from somewhere other than Prometheus
2. Istio itself, e.g. essentially making Kiali support not only Istio, whether that means pulling metrics from p8s or not, but non Istio metrics, non Istio entities, etc.
Which option is this issue about?

Additionally, whether this is option 1 or 2, (and 2 is much bigger than 1 IMO)
How would this change affect also other Kiali screens and features?
cc @lucasponce

**Comment by jshaughn** [comment]
In this context "telemetry vendor" refers to its definition in the graph impl code, which defines it as whatever backend source is providing the traffic telemetry.  The graph code defines two vender-specific impls.  One is "telemetry" and is responsible for generating a "TrafficMap".   Our sole implementation is "istio", which uses Prometheus to store its metrics.   The other is "config" and is responsible for generating the config consumed by the client.  Our sole impl is "cytoscape".

This issue can encompass anything we do to help support creation of a telemetry vendor other than Istio.  As long as the telemetry vendor generates a valid TrafficMap the config code can generate the final config.  In other words, the istio impl could be replaced with Linkerd and the resulting Trafficmap would be valid for the existing cytoscape config vendor, and therefore valid for the Kiali current UI.  Note that our graph UI is in no way generic, and is tied to Cytoscape config.

Note that the TrafficMap has been defined with istio as the reference impl, so it may tend towards istio terminology, but is still somewhat generic.  If and when someone were to actually attempt a different telemetry impl, I'm sure we'd see where we have made assumptions.  But, I think we have made good progress with providing a feasible architecture for doing so.
